src/black/__init__.py

```diff
@@ -263,7 +263,10 @@ def get_cache_key(self) -> str:
         if self.target_versions:
             version_str = ",".join(
                 str(version.value)
-                for version in sorted(self.target_versions, key=lambda v: v.value)
+                # Returning Any from function declared to return "Set[TargetVersion]"
+                for version in sorted(
+                    self.target_versions, key=lambda v: v.value  # type: ignore
+                )
```

Comment:
This is the main change I need reviewed here in case someone knows how to type this - I think field() returns Any?

Reply:
That's pretty weird, not sure what's going on here. Is it complaining about the lambda?

Reply:
I changed it to a real function and was same error too. It’s saying v.value is Any ...

Reply:
I think mypy just doesn't know the type of `v.value`. In 0.782, the return type of sorted's key was Any (https://github.com/python/typeshed/pull/4192/files). I'm assuming mypy's `--warn-return-any` doesn't complain if you return Any from a function that's meant to declare Any, so this only started getting reported when sorted's key gained a non-Any return type.

Reply:
See also https://github.com/python/mypy/issues/8722